src/utils/hooks/useDataViews.tsx

```diff
@@ -83,59 +108,37 @@ export const useDataViewsDetail = ({ dataViewCode, recordId }: DataViewDetailsPr
   } = useUserState()
 
   useEffect(() => {
-    processGetRequest(
-      getServerUrl('dataViews', { dataViewCode, itemId: recordId }),
+    processRequest({
+      url: getServerUrl('dataViews', { dataViewCode, itemId: recordId }),
       setError,
       setLoading,
-      setDataViewDetail
-    )
+      setStateMethod: setDataViewDetail,
+    })
   }, [templatePermissions, dataViewCode, recordId])
 
   return { error, loading, dataViewDetail }
 }
 
-const processGetRequest = (
-  url: string,
-  setErrorMethod: (_: ErrorResponse | null) => void,
-  setLoadingMethod: (_: boolean) => void,
+interface RequestProps {
+  url: string
+  setError: (_: ErrorResponse | null) => void
+  setLoading: (_: boolean) => void
```

Comment:
It was a dumb idea to split these into two separate methods for "get" and "post", I just did it cos I was in a hurry before. This is much nicer :)